misk/src/main/kotlin/misk/client/PeerClientFactory.kt

```diff
@@ -0,0 +1,56 @@
+package misk.client
+
+import misk.clustering.Cluster
+import misk.config.AppName
+import misk.security.cert.X500Name
+import misk.web.jetty.JettyService
+import okhttp3.OkHttpClient
+import java.security.cert.X509Certificate
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/**
+ * Factory that creates [OkHttpClient]s for connecting to another instance of the same application
+ * running in the same cluster.
+ */
+@Singleton
+class PeerClientFactory @Inject constructor(
+  @AppName private val appName: String,
+  private val jetty: JettyService,
```

Comment:
Using Jetty feels a bit weird and heavy. Injecting the web server config doesn't work in tests, though, since ports are dynamically assigned.

Reply:
lol need dat PortPicker

Reply:
wouldn't be terrible to statically assign ports in tests. We already expect in tests 2181 for ZK and 27001-4 for Vitess, etc. Though maybe the argument should be that those services should be using dynamically assigned ports in tests...

Reply:
also does something need to depend on jetty service's key in the producer/consumer graph for the port to be assigned, or does that happen earlier?